packages/volto/src/components/manage/UniversalLink/UniversalLink.jsx

```diff
@@ -109,7 +110,7 @@ const UniversalLink = ({
   } else if (isDownload) {
     tag = (
       <a
-        href={flattenToAppURL(url)}
+        href={addPrefixPath(flattenToAppURL(url))}
```

Comment:
So what if instead of modifying any invocation to `flattenToAppURL` (and I'm sure we are missing a lot), we incorporate this to it? I think it's not acceptable to modify every single apparition of it and the future ones.

Reply:
@sneridagh I encountered challenges when attempting to integrate it directly into the "flattenToAppUrl" [function](https://github.com/eea/volto-prefixpath/blob/master/src/customizations/volto/helpers/Url/Url.js#L108), primarily due to the occurrence of duplicate prefixes because we already have a router basename set in this PR. Just to let you know we have recently deployed https://water.europa.eu/freshwater which is using bits and pieces of this PR. We did some customizations which will be removed when volto 17 is integrated into that site. I'm not sure if we applied special deployment tactics maybe @olimpiurob or @claudiaifrim  knows more about this context. 

Just for people who wants to test this feature, currently for development we use
```
RAZZLE_PREFIX_PATH=/freshwater RAZZLE_INTERNAL_API_PATH=http://x.x.x.x:port/Plone yarn start

```
You might also need to set `RAZZLE_API_PATH` if your backend lives in another site than /Plone.

Reply:
@nileshgulia1 I didn't understand the difficulty of using `addPrefixPath` inside `flattenToAppURL`. Are there situations where it is not necessary to use `addPrefixPath`?

Reply:
@wesleybl Basically paths that are rendered using `<Link />` or `<UniversalLink/>` need not be prefixed. This also includes hashlinks. In volto-prefixpath we don't set the router basename as prefix, thus we use this approach. 
The image components in our addons were our major source of trouble because in volto 17 we have a single Image component and the rule is to use it everywhere to render images. So, we use this approach until we migrate all addons to volto 17.

Reply:
> I encountered challenges when attempting to integrate it directly into the "flattenToAppUrl" [function](https://github.com/eea/volto-prefixpath/blob/master/src/customizations/volto/helpers/Url/Url.js#L108), primarily due to the occurrence of duplicate prefixes because we already have a router basename set in this PR.

@nileshgulia1 did you change this behavior in [this commit](https://github.com/plone/volto/pull/3592/commits/433ad1cb272130ebfdd8b5945c08dedbcc7d68f0)?

Now `flattenToAppURL` takes care of everything?

Reply:
@nileshgulia1 with its implementation, we have a duplicate prefix in the URL, when we click, for example, on an item in the navigation menu. It would be what you said here:

> I encountered challenges when attempting to integrate it directly into the "flattenToAppUrl" [function](https://github.com/eea/volto-prefixpath/blob/master/src/customizations/volto/helpers/Url/Url.js#L108), primarily due to the occurrence of duplicate prefixes because we already have a router basename set in this PR.

Any suggestions on how to fix this?